nuxeo-drive-client/nxdrive/tests/test_engine_offline.py

```diff
@@ -49,6 +50,19 @@ def mock_execute(self, *args, **kwargs):
             raise URLError("server is offline in mock_execute:")
         return EngineOffLineTestCase.original_execute(self, *args, **kwargs)
 
+    def wait_for_upload(self, items=[], timeout=30):
+        '''
+            Wait for a list of file/folders to upload
+        '''
+        elapsed = 0
+        not_synced = items
+        while not_synced and elapsed < timeout:
+            sleep(2)
+            elapsed = elapsed + 2
+            states = [self.get_dao_state_from_engine_1(item) for item in items]
+            not_synced = [state for state in states if state and str(state.pair_state) != 'synchronized']
```

Comment:
could use the built-in function 'any', although for a small list doesn't matter:
not_synced = any(state.pair_state != 'synchronized' for state in states)
also, isn't pair_state attribute already a string?